fivetran_provider_async/operators/fivetran.py

```diff
@@ -0,0 +1,64 @@
+from typing import Any, Dict, Optional
+
+from airflow.exceptions import AirflowException
+from airflow.utils.context import Context
+from fivetran_provider.operators.fivetran import FivetranOperator
+
+from fivetran_provider_async.triggers.fivetran import FivetranTrigger
+
+
+class FivetranOperatorAsync(FivetranOperator):
+    """
+    `FivetranOperatorAsync` submits a Fivetran sync job , and polls for its status on the
+    airflow trigger.`FivetranOperatorAsync` requires that you specify the `connector_id` of
+    the sync job to start. You can find `connector_id` in the Settings page of the connector
+    you configured in the `Fivetran dashboard <https://fivetran.com/dashboard/connectors>`_.
+
+    :param fivetran_conn_id: `Conn ID` of the Connection to be used to configure
+        the hook.
+    :type fivetran_conn_id: str
+    :param connector_id: ID of the Fivetran connector to sync, found on the
+        Connector settings page in the Fivetran Dashboard.
+    :type connector_id: str
+    :param poke_interval: Time in seconds that the job should wait in
+        between each tries
+    :type poke_interval: int
+    :param fivetran_retry_limit: # of retries when encountering API errors
+    :type fivetran_retry_limit: Optional[int]
+    :param fivetran_retry_delay: Time to wait before retrying API request
+    :type fivetran_retry_delay: int
+    """
+
+    def execute(self, context: Dict[str, Any]) -> None:
+        """Start the sync using synchronous hook"""
+        hook = self._get_hook()
+        hook.prep_connector(self.connector_id, self.schedule_type)
+        hook.start_fivetran_sync(self.connector_id)
+
+        """Defer and poll the sync status on the Trigger"""
+        self.defer(
+            timeout=self.execution_timeout,
+            trigger=FivetranTrigger(
+                task_id=self.task_id,
+                fivetran_conn_id=self.fivetran_conn_id,
+                connector_id=self.connector_id,
+                polling_period_seconds=self.poll_frequency,
+            ),
+            method_name="execute_complete",
+        )
+
+    def execute_complete(self, context: "Context", event: Optional[Dict[Any, Any]] = None) -> None:
+        """
+        Callback for when the trigger fires - returns immediately.
+        Relies on trigger to throw an exception, otherwise it assumes execution was
+        successful.
+        """
+        if event:
+            if "status" in event and event["status"] == "error":
+                msg = "{0}: {1}".format(event["status"], event["message"])
+                raise AirflowException(msg)
+            if "status" in event and event["status"] == "success":
+                self.log.info(
+                    event["message"],
+                )
```

Comment:
```suggestion
                self.log.info("Task succeeded. Status: %s", event["message"])
```

Reply:
We get event["message"] as below. Hence retaining the code as is, since the below message is more relevant.
```
msg = "Fivetran connector %s finished syncing at %s" % (
                        self.connector_id,
                        self.previous_completed_at,
                    )
```

Reply:
Logs below for reference

```
[2022-11-30, 07:16:59 UTC] {taskinstance.py:1383} INFO - Executing <Task(FivetranOperatorAsync): fivetran_async_op> on 2022-11-29 07:15:51.029638+00:00
[2022-11-30, 07:16:59 UTC] {standard_task_runner.py:55} INFO - Started process 647 to run task
[2022-11-30, 07:16:59 UTC] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'example_fivetran_only_operator_async', 'fivetran_async_op', 'scheduled__2022-11-29T07:15:51.029638+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/example_fivetran_only_operator_async.py', '--cfg-path', '/tmp/tmp1y2bk57s']
[2022-11-30, 07:16:59 UTC] {standard_task_runner.py:83} INFO - Job 4: Subtask fivetran_async_op
[2022-11-30, 07:16:59 UTC] {task_command.py:376} INFO - Running <TaskInstance: example_fivetran_only_operator_async.fivetran_async_op scheduled__2022-11-29T07:15:51.029638+00:00 [running]> on host 40af35e92f3c
[2022-11-30, 07:16:59 UTC] {operators.py:54} INFO - Fivetran connector bronzing_largely finished syncing at 2022-11-30T07:16:46.877000+00:00
```